frontend/views/pages/Contributions.vue

```diff
@@ -208,28 +187,77 @@ export default {
       }
     }
   },
-  validations: {
-    form: {
-      incomeAmount: {
-        [L('field is required')]: required,
-        [L('cannot be negative')]: v => v >= 0,
-        [L('cannot have more than 2 decimals')]: decimals(2)
-      },
-      pledgeAmount: { required, minValue: v => v >= 0, decimals: decimals(2) }
-    }
-  },
   computed: {
     ...mapGetters([
+      'ourUsername',
       'groupSettings',
+      'groupMembersCount',
       'groupProfile',
+      'groupProfiles',
+      'thisMonthsPayments',
       'groupMincomeFormatted',
-      'ourUsername'
+      'groupIncomeDistribution'
     ]),
-    doesReceiveMonetary () {
-      return !!this.fakeStore.receiving.monetary && !this.ephemeral.isEditingIncome
+    memberGroupProfile () {
+      return this.groupProfile(this.ourUsername) || {}
+    },
+    upTo () {
+      const amount = this.memberGroupProfile[this.memberGroupProfile.incomeDetailsType]
+      if (!amount) return false
+      return this.currency.displayWithCurrency(this.needsIncome ? this.groupSettings.mincomeAmount - amount : amount)
+    },
+    doesReceive () {
+      return this.needsIncome || this.receivingNonMonetary
+    },
+    doesGive () {
+      return this.hasPayments || this.memberGroupProfile.nonMonetaryContributions
+    },
+    needsIncome () {
+      return this.memberGroupProfile.incomeDetailsType === 'incomeAmount'
+    },
+    hasPayments () {
+      return Object.keys(this.givingMonetary.list).length > 0
+    },
+    currency () {
+      return currencies[this.groupSettings.mincomeCurrency]
+    },
+    distribution () {
+      return this.thisMonthsPayments.frozenDistribution || this.groupIncomeDistribution
+    },
+    receivingNonMonetary () {
+      const groupProfiles = this.groupProfiles
+      return Object.keys(groupProfiles)
+        .filter(key => key !== this.ourUsername && groupProfiles[key].nonMonetaryContributions)
```

Comment:
Check for `&& groupProfiles[key].nonMonetaryContributions` isn't needed, that is always defined

Reply:
I check the length now